src/schema/decimal.rs

```diff
@@ -0,0 +1,84 @@
+use crate::utils;
+use bigdecimal::BigDecimal;
+use num_bigint::ToBigInt;
+use std::io::{Read, Write};
+use std::str::FromStr;
+use yaserde::{YaDeserialize, YaSerialize};
+
+#[derive(Default, PartialEq, Debug)]
+pub struct Decimal {
+    pub value: BigDecimal,
+}
+
+impl Decimal {
+    fn from_bigdecimal(bigdecimal: BigDecimal) -> Self {
+        Decimal{value: bigdecimal}
+    }
+
+    fn to_bigdecimal(&self) -> Option<BigDecimal> {
+        Some(self.value.clone())
+    }
+}
+
+impl YaDeserialize for Decimal {
+    fn deserialize<R: Read>(reader: &mut yaserde::de::Deserializer<R>) -> Result<Self, String> {
+        utils::yaserde::deserialize(reader, |s| Ok(Decimal::from_bigdecimal(BigDecimal::from_str(s).unwrap())))
```

Comment:
Here we can avoid unwrapping using something like 
```rust
utils::yaserde::deserialize(reader, |s| {
    BigDecimal::from_str(s)
        .map(Decimal::from_bigdecimal)
        .map_err(|e| e.to_string())
})
```

Reply:
Done